### PR TITLE
Pseudo code to show the flow to fetch holdings from a host record

### PR DESCRIPTION
**Pull request by hectorcorrea**
Prototype to address #2431

This is a rather crude proof of concept to show how we could detect that we need to fetch holdings from a host record. YThe basic idea is to (1) detect that this record is container in another, (2) fetch the data from the host record, and (3) merge the holdings from the host into the current record. 

Only the first part is shown in this draft PR.

Unfortunately the field that indicates the host record is currently not being populated because [we are indexing the wrong field](https://github.com/pulibrary/bibdata/blob/alma/marc_to_solr/lib/traject_config.rb#L586) (we are indexing `773$w` but we need `773$x`) so the logic is hard-coded but it shows the general idea.

### CODE REVIEW

**Comment by hackartisan** [comment]
Hm, the description of the PR says we are indexing `w` but want `x`, and the code comment says we are indexing `x` but want `w`. I am pretty sure `w` is what we want, right?

This seems good and we would still need to figure out the search results availability piece.

File: app/controllers/catalog_controller.rb [hunk 1]
```diff
@@ -673,4 +673,26 @@ def index
...
+
+  def show
+    # Let Blacklight do the normal flow
+    value = super
+
+    host_id = nil
+    if @document.id == "9929455793506421" || @document["contained_in_s"]
+      # This value should come from @document["contained_in_s"] but right now the value
+      # it's empty because we are indexing subfield x instead of subfield w.
+      host_id = "9929455793506421" || @document["contained_in_s"]
```
Comment (christinach) [inline app/controllers/catalog_controller.rb:685]: We are indexing 773w. https://github.com/pulibrary/bibdata/blob/35e81e844b6491cdedd65f20313838364eaf31b7/marc_to_solr/lib/traject_config.rb#L586

**Comment by christinach** [comment]
I think that records that have 773x were part of a test. @mzelesky can provide more input on that.

**Comment by mzelesky** [comment]
Yes, that is the old version of a record. There is a new version of that record that uses the final structure. The subfield x was a mistake.

**Comment by hectorcorrea** [comment]
Great, thanks for the clarification @christinach @mzelesky

**Comment by hectorcorrea** [comment]
Replaced with https://github.com/pulibrary/orangelight/pull/2553

**Pull request closed without merging**